### PR TITLE
combine and report unittest results

### PR DESCRIPTION
**Pull request by lucienwang1009**
Merge all unittests cross different platform, different TF version, different python version, etc.
You can find the unittest coverage report in `combine_and_report_coverage` job of unittest build.
Report of html is also available as a build artifact.

### CODE REVIEW

**Comment by nbcsm** [comment]
seems something is wrong with the test run name?

image

**Comment by lucienwang1009** [comment]
@nbcsm azure pipelines will automatically add job_name ahead of its title. 
We didn't set it before so the title was what we set in config.
Now we need these unittest jobs to run before the report job and thus assign it explicitly.

**Comment by nbcsm** [comment]
can we leverage stage?
https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema?view=azure-devops&tabs=schema#stage

**Comment by lucienwang1009** [comment]
@nbcsm look like fitting to our scenario, I'll give a try.

File: ci_build/azure_pipelines/onnxruntime_nightly_test.yml [hunk 1]
```diff
@@ -12,3 +12,8 @@ jobs:
       - template: 'unit_test.yml'
         parameters:
             onnx_opsets: ['10', '9', '8', '7']
+    report_unittest: 'True'
+
+- template: 'templates/combine_test_coverage.yml'
+  parameters:
+    combined_jobs: ['nightly']
```
Comment (nbcsm) [inline ci_build/azure_pipelines/onnxruntime_nightly_test.yml]: combined_jobs?

File: ci_build/azure_pipelines/templates/job_generator.yml [hunk 2]
```diff
@@ -9,73 +9,83 @@ parameters:
   onnx_backends: {onnxruntime: ['0.4.0']}
   job: {}
   run_setup: 'True'
+  report_unittest: 'False'
```
Comment (nbcsm) [inline ci_build/azure_pipelines/templates/job_generator.yml]: rename to report_coverage?

**Pull request merged**